Mitigate shadowing of logging module

cybit opened the pull request:
This relates to and should FIX #755  .
The way I see it it should fix #298 too.
Problem: Because of the shadowing you will get an exception whenever the warning or error would have been printed.
Python Interpreter: 2.7, 3.7
jira-python: 2.0.0 and older
OS: Linux Debian, Linux Ubuntu

Stacktrace:

> Exception AttributeError: "Magic instance has no attribute 'cookie'" in <bound method Magic.__del__ of <magic.Magic instance at 0x7fe763bc4830>> ignored
> WARNING:root:Bug https://jira.atlassian.com/browse/JRA-59676 trying again...
> WARNING:root:Bug https://jira.atlassian.com/browse/JRA-59676 trying again...
> WARNING:root:Bug https://jira.atlassian.com/browse/JRA-59676 trying again...
> Traceback (most recent call last):
>   File "./test.py", line 51, in <module>
> [...]
>   File "/usr/local/lib/python2.7/dist-packages/jira/client.py", line 476, in __init__
>     logging.error("invalid server_info: %s", si)
> AttributeError: 'bool' object has no attribute 'error'
> 

What I did in the pull request:
604e9b1 
* I assumed that you want to follow https://docs.python.org/3/howto/logging.html#library-config
* I changed the Interface for JIRA() so that logging defaults to None. 
* Tried to convey that logging is deprecated
* If the call is JIRA() or JIRA(logging=Null) or JIRA(logging=False) I reimport logging and repeat the line with logging.NullHandler()
* If the call is JIRA(logging=True) I reimport logging but with the default logger.
* I removed one last line that used self.logging and removed the variable 

2caecbe 
* Fixed up tests.py so that JIRA() will no longer be called with the attribute 
* Did some reformatting while at it. 

Testint:
Tox did fail.
[test-result.txt](https://github.com/pycontribs/jira/files/2845249/test-result.txt)

Main reasons where 
* AttachmentTests.test_0_attachment_meta
* JiraShellTests.test_jirashell_command_exists
* pycontribs.atlassian.net does not respond as expected
* sphinx and dependencies were not installed in pip

hdost commented on tests/tests.py:
```diff
@@ -1995,10 +2003,7 @@ class OtherTests(unittest.TestCase):
-                 basic_auth=("xxx", "xxx"),
-                 validate=True,
-                 logging=False)
+            JIRA('https://jira.atlassian.com', basic_auth=("xxx", "xxx"), validate=True, )
```
remove trailing comma.

cybit replied:
Removed in 38546b4 as requested

cybit commented:
FYI: Travis Check Failed because of 

> 3.5 is not installed; attempting download
Downloading archive: https://travis-ci-python-archives.global.ssl.fastly.net/binaries/ubuntu/14.04/x86_64/python-3.5.tar.bz2
360.96s$ curl -sSf -o python-3.5.tar.bz2 ${archive_url}
curl: (56) SSL read: error:00000000:lib(0):func(0):reason(0), errno 104
Unable to download 3.5 archive. The archive may not exist. Please consider a different version.

cybit commented:
@hdost  What was missing after I removed the trailing comma as per your request?
And why do I get a 401 now after I did a git rebase and fixed the merge conflict in code that I did not even touch?
